src/app/client/src/app/modules/program-dashboard/components/program-datasets/program-datasets.component.ts

```diff
@@ -304,43 +299,35 @@ export class DatasetsComponent implements OnInit {
   }
 
   districtSelection($event){
-    this.districtId = $event.value;
     this.reportForm.controls.districtName.setValue($event.value);
   }
 
   organisationSelection($event){
-    this.organisationId = $event.value;
     this.reportForm.controls.organisationName.setValue($event.value);
   }
 
   reportChanged(selectedReportData) {
     this.selectedReport = selectedReportData;
-    this.filter = this.selectedReport['filters'];
-  }
- addFilters(){
-  if(this.districtId == undefined){
-    this.filter = this.filter.filter( element => {
-      return element.dimension != 'district_externalId'
-    })
   }
-  this.filter.forEach((data, index) => {
-    if(data.dimension == 'program_id'){
-      data.value = this.programSelected;
-    }
-    if(data.dimension == 'solution_id'){
-      data.value = this.reportForm.controls.solution.value;
-    }
-    if(this.districtId !== undefined && data.dimension == 'district_externalId'){
-      data.value = this.districtId;
-    }
-    if(this.organisationId !== undefined && data.dimension == 'organisation_id' ){
-      data.value = this.organisationId;
+ addFilters(){ 
+    let filterKeysObj = {
+    program_id:this.reportForm.controls.programName.value,
```

Comment:
if one of the object is not there the flow brakes with undefined issue please use the _.get function

Reply:
Sure will do, thank you